src/IssueViz/Editor/LocationTagging/LocationTagger.cs

```diff
@@ -29,8 +29,6 @@
 
 namespace SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging
 {
-    // TODO: track document renames: https://github.com/SonarSource/sonarlint-visualstudio/issues/1662
```

Comment:
This tagger is storing the file path at the point it is created, and using that value when calling the location service. That works partially at the moment because the snapshot file name is not being updated currently. However, the tagger won't work properly in the following scenario:
* open a file -> analysed ok, tagger created
* rename the file while it is open -> snapshot is refreshed; it knows about both old and new names
* edit the doc -> the tagger will still update correctly at this point. It's using the old file name, but it works because the snapshot still has the old name in its list of affected files
* save the file -> re-analysis -> new snapshot -> tagger told about the new issues.
At this point, the tagger only knows about the old name, and the new snapshot only knows about the new name.

The tagger is no longer tracking the latest issues, and any notifications it sends to the location store are ignored because they are using the old file name.

Reply:
FYI I pushed another commit in a branch called `RG/1662-file-renames_DP1` that changes the tagger so it doesn't ever store the file path (it fetches it from the document each time). However, it's only part of the solution, and I'm not sure it's the right approach.